### ISSUE TITLE
KeyAttributeLookup entries should be named by path

### ISSUE THREAD

**Issue by HunterMcGushion** [HunterMcGushion/hyperparameter_hunter#75]
- Execute `examples/lib_keras_example.py`
- Loss functions are saved to ‘examples/HyperparameterHunterAssets/KeyAttributeLookup/0.json’
    - This is extremely ambiguous and likely to cause conflicts if another parameter inside a list is also hashed
    - Need to name files according to path to the hyperparameter, not just the hyperparameter key
- **See :meth:`hyperparameter_hunter.key_handler.KeyMaker.add_complex_type_lookup_entry`** for todo comment and for probable location of the necessary changes
